### PR TITLE
Improving responsive image support - update

### PR DESCRIPTION
**Pull request by tevko**
Hey @iamtakashi here's an overview of the updates since the last PR:
- Version numbers have been fixed
- Extra image size moved to twentysixteen_setup
- 840px wide image in pages and post content accounted for 
- Custom header size fixed, only 1200px wide
- add_filter moved to correct location in funtions.php 
- Travis formatting issues corrected
- We have a MQ value in the `sizes` attribute that's larger than 1320px, but that's because I noticed that the image can still get larger after the browser is wider than 1320px if it's a full size image. It's only a difference of 59px but I left it in for the sake of precision

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -72,6 +72,13 @@ function twentysixteen_setup() {
 	add_theme_support( 'post-thumbnails' );
 	set_post_thumbnail_size( 1200, 0, true );
 
+	/*
+	 * Add image size to enhance responsive image functionality
+	 *
+	 * @link https://developer.wordpress.org/reference/functions/add_image_size/
+	 */
+	add_image_size( 'intermediate-width', 600, 600 );
```
Comment (grappler) [inline functions.php]: If the image size handle is not prefixed then there is a chance that a plugin could define the same name. The two possibilities would be to name it `twentysixteen-intermediate-width` or `600x600`. Further reading: https://github.com/grappler/wp-standard-handles#image-sizes

Comment (tevko) [inline functions.php]: @grappler thanks for the heads up! This has been updated

**Comment by joemcgill** [comment]
A quick note on this PR: there may be a slight change to the filter for `wp_get_attachment_image_sizes()` soon (see Trac [34379](https://core.trac.wordpress.org/ticket/34379)), in which case the filter should be rewritten like this:

```
/**
 * Add custom image sizes attribute to enhance responsive image functionality
 *
 * @since Twenty Sixteen 1.0
 *
 * @param string       $sizes A source size value for use in a 'sizes' attribute.
 * @param int          $id    Post ID of the original image.
 * @param array|string $size  Image size. Accepts any valid image size, or an array of width and 
 *                            height values in pixels (in that order). Default 'medium'.
 * @return string A source size value for use in a 'sizes' attribute.
 */
function twentysixteen_image_sizes_attr( $sizes, $id, $size ) {
    if ( 'large' === $size ) {
        $sizes = '(max-width: 709px) 85vw, (max-width: 909px) 66vw, (max-width: 984px) 60vw, (max-width: 1361px) 44vw, 600px';
    } elseif ( 'full' === $size || 'post-thumbnail' === $size ) {
        $sizes = '(max-width: 709px) 85vw, (max-width: 909px) 66vw, (max-width: 984px) 60vw, (max-width: 1362px) 62vw, 840px';
    }
    return $sizes;
}
add_filter( 'wp_get_attachment_image_sizes', 'twentysixteen_image_sizes_attr', 10 , 3 );
```

**Comment by tevko** [comment]
Keeping eyes on this patch, will update the PR once approved

**Comment by karmatosed** [comment]
Can we get a refresh on this patch?

**Comment by tevko** [comment]
@karmatosed will update later tonight!

**Comment by tevko** [comment]
@karmatosed updated!

**Comment by iamtakashi** [comment]
@tevko, thanks for the update. I'll test it through tomorrow.

**Comment by iamtakashi** [comment]
Sorry @tevko, it's now got out of sync again because of the last update in CONTRIBUTORS.md. Could you update the PR again and remove your name from the file so that it's the same as the latest version? 

We'll take care of adding your name when it's merged.

**Comment by tevko** [comment]
@iamtakashi  Sorry about that! PR updated!

**Comment by iamtakashi** [comment]
Tevko and I had a chat on Slack but I'll leave some notes here for the record.
- OSs and browsers handle differently sub-pixel rendering but let's change the max breakpoint to be @1362px because that's when the large size and the header image are supposed to be 600px and 1200px.
- Assuming we're going to have medium_large image size in core (https://core.trac.wordpress.org/ticket/34196), let's remove the `twentysixteen-intermediate-width` size.

**Comment by iamtakashi** [comment]
The updated PR looks great and now tevko is working on no-widgets layout where the size of featured image is larger (1200px wide).

**Comment by tevko** [comment]
@iamtakashi PR updated to take pages and no-sidebar layouts into account

**Comment by iamtakashi** [comment]
We're waiting for https://core.trac.wordpress.org/ticket/34477.

Also, we're going to take a filter approach for the one for a custom header so that child themes can adjust easily.

**Comment by iamtakashi** [comment]
Thanks for the update, @tevko! I've realised that we need more adjustments but I'd like to get this in for now so that we can move this forward. I'm going to work on further improvement.

**Pull request merged**